### PR TITLE
added parent pom and commons to move common code in another repository

### PR DESCRIPTION
**Pull request by ayansen**

### CODE REVIEW

File: commons/src/main/scala/com/expedia/www/haystack/trends/commons/entities/MetricPoint.scala [hunk 1]
```diff
@@ -54,6 +54,7 @@ object TagKeys {
   val RESULT_KEY = "result"
   val STATS_KEY = "stat"
   val ERROR_KEY = "error"
+  val INTERVAL_KEY = "interval"
```
Comment (shellbj) [inline commons/src/main/scala/com/expedia/www/haystack/trends/commons/entities/MetricPoint.scala]: Should this be buried in commit about moving and pom updates?

File: span-timeseries-transformer/pom.xml [hunk 2]
```diff
@@ -221,169 +76,31 @@
...
-                <executions>
-                    <execution>
-                        <id>compile-scalastyle</id>
-                        <goals>
-                            <goal>check</goal>
-                        </goals>
-                    </execution>
-                </executions>
+                <groupId>org.scalatest</groupId>
+                <artifactId>scalatest-maven-plugin</artifactId>
```
Comment (shellbj) [inline span-timeseries-transformer/pom.xml]: duplicate plugin; did you mean scalastyle?

File: timeseries-aggregator/src/main/scala/com/expedia/www/haystack/trends/kstream/serde/metric/MetricSerde.scala [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.expedia.www.haystack.trends.kstream.serde.metric
```
Comment (shellbj) [inline timeseries-aggregator/src/main/scala/com/expedia/www/haystack/trends/kstream/serde/metric/MetricSerde.scala]: Do we want to add new things is a reorg commit?

File: .gitignore [hunk 4]
```diff
@@ -7,5 +7,4 @@ target/
 lib/
 logs/
 **/.idea/
-/timeseries-aggregator/haystack-timeseries-aggregator.iml
-span-timeseries-transformer/haystack-span-timeseries-transformer.iml
+**/*.iml
```
Comment (shellbj) [inline .gitignore]: `*.iml` works just fine

File: .travis.yml [hunk 5]
```diff
@@ -8,15 +8,12 @@ env:
...
 script:
   # build, create docker image
   # upload to dockerhub only for master(non PR) and tag scenario
-  - if ([ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]) || [ -n "$TRAVIS_TAG" ];  then cd "$SUB_PROJECT" && make release; else cd "$SUB_PROJECT" && make all; fi
+  - if ([ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]) || [ -n "$TRAVIS_TAG" ];  then make release; else make all; fi
 
 notifications:
   email:
-    - haystack-notifications@expedia.com
+    - haystack-notifications@expedia.com
```
Comment (shellbj) [inline .travis.yml]: new line

File: commons/commons.iml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="false">
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (shellbj) [inline commons/commons.iml]: new line

File: commons/commons.iml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shellbj) [inline commons/commons.iml]: remove this file

File: commons/haystack-trends-commons.iml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shellbj) [inline commons/haystack-trends-commons.iml]: remove this file

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+                    <version>${protobuf-maven-plugin.version}</version>
+                    <executions>
+                        <execution>
+                            <phase>generate-sources</phase>
+                            <goals>
+                                <goal>run</goal>
+                            </goals>
+                            <configuration>
+                                <includeDirectories>
+                                    <include>${project.basedir}/../haystack-idl/proto</include>
```
Comment (shellbj) [inline pom.xml]: `${project.basedir}/../haystack-idl/proto` is outside of this project

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+                        <phase>process-test-resources</phase>
+                        <goals>
+                            <goal>testCompile</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (shellbj) [inline pom.xml]: new line

**Comment by shellbj** [comment]
Consider adding the maven-release-plugin and the `scm` sections to the various poms.

File: timeseries-aggregator/src/main/scala/com/expedia/www/haystack/trends/kstream/serde/metric/MetricSerde.scala (hunk 3 above)
Comment (ayansen) [reply timeseries-aggregator/src/main/scala/com/expedia/www/haystack/trends/kstream/serde/metric/MetricSerde.scala]: this was there, its just a reorg

File: commons/src/main/scala/com/expedia/www/haystack/trends/commons/entities/MetricPoint.scala (hunk 1 above)
Comment (ayansen) [reply commons/src/main/scala/com/expedia/www/haystack/trends/commons/entities/MetricPoint.scala]: it was there in the aggregator metric point definitition

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,394 @@
...
+        <pegdown.version>1.6.0</pegdown.version>
+        <scalatest.version>3.0.3</scalatest.version>
+        <slf4j.version>1.7.25</slf4j.version>
+        <kafka.version>0.11.0.0</kafka.version>
+        <junit.version>4.12</junit.version>
+        <typesafe-config.version>1.3.1</typesafe-config.version>
+        <logback.version>1.2.3</logback.version>
+        <metrics-core.version>3.0.2</metrics-core.version>
+        <protobuf.version>3.3.1</protobuf.version>
+        <protobuf-maven-plugin.version>3.3.0.1</protobuf-maven-plugin.version>
```
Comment (shellbj) [inline pom.xml]: move this with the other plugin versions

File: pom.xml (hunk 9 above)
Comment (shellbj) [reply pom.xml]: `${project.parent.basedir}/haystack-idl/proto` works however, if we don't add sub-sub modules

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,394 @@
...
+                                <htmlreporters>${project.build.directory}/scalatest/html</htmlreporters>
+                            </configuration>
+                        </execution>
+                    </executions>
+                </plugin>
+
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-jar-plugin</artifactId>
+                    <version>3.0.2</version>
```
Comment (shellbj) [inline pom.xml]: You have a pinned version and a variable for the same; both are out of sync.

File: span-timeseries-transformer/pom.xml (hunk 2 above)
Comment (ayansen) [reply span-timeseries-transformer/pom.xml]: fixed

File: pom.xml (hunk 12 above)
Comment (ayansen) [reply pom.xml]: done

File: pom.xml (hunk 11 above)
Comment (ayansen) [reply pom.xml]: done

**Pull request merged**